### ISSUE TITLE
Lektor Sitemaps incompatible with Jinja2 2.9

### ISSUE THREAD

**Issue by freakboy3742** [lektor/lektor#343]
Using Lektor 2.3; If you have a project with a sitemap (using the template suggested by the Lektor docs):
```
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  {%- for page in [site.root] if page != this recursive %}
  <url><loc>{{ page|url(external=true) }}</loc></url>
  {{- loop(page.children) }}
  {%- endfor %}
</urlset>
```

and you have jinja2 2.9 - 2.9.3, in your environment, you get a build error:
```
(lektor) hammer:pybee.org rkm$ lektor build
Started build
U ar_AR/sitemap.xml
E ar_AR/sitemap.xml (NameError: free variable 'l_1_this' referenced before assignment in enclosing scope)
U sitemap.xml
E sitemap.xml (NameError: free variable 'l_1_this' referenced before assignment in enclosing scope)
U pt_BR/sitemap.xml
E pt_BR/sitemap.xml (NameError: free variable 'l_1_this' referenced before assignment in enclosing scope)
U zh_TW/sitemap.xml
E zh_TW/sitemap.xml (NameError: free variable 'l_1_this' referenced before assignment in enclosing scope)
```

Downgrading to Jinja2 2.8.1 seems to resolve the problem.

I'm guessing either the docs need to be updated to reflect a required change in the sitemap template, or there's a bug in Jinja2.

**Comment by singingwolfboy** [comment]
Huh, that's interesting. I'm guessing that this is a bug in Jinja2, but I don't really know. Can you [create an issue on the Jinja2 repo](https://github.com/pallets/jinja/issues), and link to this issue from that one?

**Comment by freakboy3742** [comment]
Done - See pallets/jinja2#649

**Comment by tariquesani** [comment]
@freakboy3742 since the bug was closed in Jinja2 2.9 are you still experiencing this issue?

**Comment by freakboy3742** [comment]
Looks like the problem has been fixed.
